orca-core/src/main/groovy/com/netflix/spinnaker/orca/pipeline/util/ContextParameterProcessor.groovy

```diff
@@ -79,16 +81,20 @@ class ContextParameterProcessor {
   }
 
   Map<String, Object> precomputeValues(Map<String, Object> context) {
-    if (context.trigger?.parameters) {
-      context.parameters = context.trigger.parameters
+    Trigger trigger = context.trigger
+    if (trigger?.parameters) {
+      context.parameters = trigger.parameters
     }
 
-    context.scmInfo = context.buildInfo?.scm ?: context.trigger?.buildInfo?.scm ?: null
+    context.scmInfo = context.buildInfo?.scm
+    if (!context.scmInfo && trigger instanceof JenkinsTrigger) {
+      context.scmInfo = trigger.buildInfo?.scm
+    }
     if (context.scmInfo && context.scmInfo.size() >= 2) {
       def scmInfo = context.scmInfo.find { it.branch != 'master' && it.branch != 'develop' }
       context.scmInfo = scmInfo ?: context.scmInfo?.first()
     } else {
-      context.scmInfo = context.scmInfo?.first()
+      context.scmInfo = context.scmInfo?.size() > 0 ? context.scmInfo.first() : null
```

Comment:
Bester if `!context.scmInfo?.isEmpty()` instead of `size`

Reply:
Except if `context.scmInfo` is `null`.